File: app/src/main/java/com/lofts/dictionary/ui/InputActivity.java
package com.lofts.dictionary.ui;

import android.content.Intent;
import android.os.Build;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.lofts.dictionary.R;
import com.lofts.dictionary.utils.RequestManager;

import java.util.HashMap;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnEditorAction;

public class InputActivity extends AppCompatActivity {

    private static final String TAG = InputActivity.class.getSimpleName();

    @BindView(R.id.edittext_input)
    EditText mEtInput;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);

        setContentView(R.layout.activity_input);

        ButterKnife.bind(this);

    }

    @OnEditorAction(R.id.edittext_input)
    public boolean setOnEditerAction(TextView v, int actionId, KeyEvent event) {
        if (actionId == EditorInfo.IME_ACTION_SEARCH || (event != null && event.getKeyCode() == KeyEvent.KEYCODE_ENTER)) {
            getTranslateResult();
            return true;
        }
        return false;
    }

    private void getTranslateResult() {
        String inputValue = mEtInput.getText().toString();
        if (TextUtils.isEmpty(inputValue)) {
            Toast.makeText(this, "请输入需要翻译的单词", Toast.LENGTH_LONG).show();
            return;
        }

        Intent intent = new Intent();
        intent.putExtra("word", inputValue);
        intent.setClass(this, ResultActivity.class);
        startActivity(intent);
    }


}

File: app/src/main/java/com/lofts/dictionary/ui/InputActivity.kt
package com.lofts.dictionary.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import android.view.KeyEvent
import android.view.Window
import android.view.WindowManager
import android.view.inputmethod.EditorInfo
import android.widget.TextView
import com.lofts.dictionary.R
import kotlinx.android.synthetic.main.activity_input.*
import org.jetbrains.anko.longToast
import org.jetbrains.anko.startActivity

/**
 * PACKAGE_NAME：com.lofts.dictionary.ui
 * DATE：2018-12-18 22:49
 * USER: asus
 * DESCRIBE:
 */
class InputActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)

        setContentView(R.layout.activity_input)

        initView()
    }

    private fun initView() {
        edittext_input.setOnEditorActionListener(TextView.OnEditorActionListener { view, actionId, event ->
            if (actionId == EditorInfo.IME_ACTION_SEARCH || event != null && event.keyCode == KeyEvent.KEYCODE_ENTER) {
                getTranslateResult()
                return@OnEditorActionListener true
            }
            return@OnEditorActionListener false
        })
    }

    private fun getTranslateResult() {
        val inputValue = edittext_input.text.toString()
        if (TextUtils.isEmpty(inputValue)) {
            longToast("请输入要查询的单词")
            return
        }
        startActivity<ResultActivity>("word" to inputValue)

    }

}